sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryParallelReadTableSource.java

```diff
@@ -41,36 +33,15 @@
 import org.apache.beam.sdk.options.ValueProvider;
 import org.apache.beam.sdk.options.ValueProvider.NestedValueProvider;
 import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A {@link BoundedSource} for reading BigQuery tables using the BigQuery parallel read API.
  */
-class BigQueryParallelReadTableSource<T> extends BoundedSource<T> {
+class BigQueryParallelReadTableSource<T> extends BigQueryParallelReadSourceBase<T> {
 
-  /**
-   * The maximum number of readers which will be requested when creating a BigQuery read session as
-   * part of a split operation, regardless of the desired bundle size.
-   */
-  private static final int MAX_SPLIT_COUNT = 10000;
-
-  /**
-   * The minimum number of readers which will be requested when creating a BigQuery read session as
-   * part of a split operation, regardless of the desired bundle size. Note that the source may
-   * still be split into fewer than ten component sources depending on the number of read locations
-   * returned by the server at session creation time.
-   */
-  private static final int MIN_SPLIT_COUNT = 10;
```

Comment:
Why do we have a min split count of 10? I am wondering for query results, should it be even smaller, since query results are usually small (of course, table view can be large).

Reply:
Hmm... the min/max split count was added at Eugene's request in the initial round of code review for the table-based source for sanity checking purposes, since Dataflow's calculations are sometimes off significantly and we want some kind of lower bound on pipeline behavior even in the worst case.

I don't actually know how to characterize the size of the average query request vs. the size of the average table read -- intuitively, I'd expect the query results to be smaller, as you say, but it's possible that they're actually larger in the case of e.g. large join results which are used as input for a Dataflow pipeline. Maybe we can find a way to gather data on this somehow.

Reply:
OK, let's leave it for Eugune then.